I'm seeing some build failures:
```
error: builder for '/nix/store/cz0ng75yp2xvcdm9pfjpq3z5clr6panz-colmap-3.7.drv' failed with exit code 2;
       last 10 log lines:
       > /nix/store/m2vh2ny7bqpwij1gpmvl5gxj7y4dgr4f-binutils-2.38/bin/ld: (.text+0x5d8a): undefined reference to `std::__throw_bad_array_new_length()'
       > /nix/store/m2vh2ny7bqpwij1gpmvl5gxj7y4dgr4f-binutils-2.38/bin/ld: /nix/store/22wgdfnkh18gva3aw2ymfzyb44z1pfrc-boost-1.77.0/lib/libboost_program_options-mt-x64.a(cmdline.o): in function `boost::program_options::detail::cmdline::run()':
       > (.text+0x6c76): undefined reference to `std::__throw_bad_array_new_length()'
       > /nix/store/m2vh2ny7bqpwij1gpmvl5gxj7y4dgr4f-binutils-2.38/bin/ld: (.text+0x6c84): undefined reference to `std::__throw_bad_array_new_length()'
       > /nix/store/m2vh2ny7bqpwij1gpmvl5gxj7y4dgr4f-binutils-2.38/bin/ld: /nix/store/22wgdfnkh18gva3aw2ymfzyb44z1pfrc-boost-1.77.0/lib/libboost_program_options-mt-x64.a(cmdline.o):(.text+0x7ab2): more undefined references to `std::__throw_bad_array_new_length()' follow
       > /nix/store/m2vh2ny7bqpwij1gpmvl5gxj7y4dgr4f-binutils-2.38/bin/ld: /nix/store/33fy01bvz2s4fq7nb536mbvcy1ifdknz-glog-0.6.0/lib/libglog.so: undefined reference to `std::__throw_bad_array_new_length()@GLIBCXX_3.4.29'
       > collect2: error: ld returned 1 exit status
       > make[2]: *** [src/exe/CMakeFiles/colmap_exe.dir/build.make:251: src/exe/colmap] Error 1
       > make[1]: *** [CMakeFiles/Makefile2:775: src/exe/CMakeFiles/colmap_exe.dir/all] Error 2
       > make: *** [Makefile:136: all] Error 2
       For full logs, run 'nix log /nix/store/cz0ng75yp2xvcdm9pfjpq3z5clr6panz-colmap-3.7.drv'.
error: builder for '/nix/store/w6gr8cjy42n5y544nxpfx9vcasm2k6aw-colmap-3.7.drv' failed with exit code 2;
       last 10 log lines:
       > (.text+0x5d80): undefined reference to `std::__throw_bad_array_new_length()'
       > /nix/store/m2vh2ny7bqpwij1gpmvl5gxj7y4dgr4f-binutils-2.38/bin/ld: (.text+0x5d8a): undefined reference to `std::__throw_bad_array_new_length()'
       > /nix/store/m2vh2ny7bqpwij1gpmvl5gxj7y4dgr4f-binutils-2.38/bin/ld: /nix/store/22wgdfnkh18gva3aw2ymfzyb44z1pfrc-boost-1.77.0/lib/libboost_program_options-mt-x64.a(cmdline.o): in function `boost::program_options::detail::cmdline::run()':
       > (.text+0x6c76): undefined reference to `std::__throw_bad_array_new_length()'
       > /nix/store/m2vh2ny7bqpwij1gpmvl5gxj7y4dgr4f-binutils-2.38/bin/ld: /nix/store/22wgdfnkh18gva3aw2ymfzyb44z1pfrc-boost-1.77.0/lib/libboost_program_options-mt-x64.a(cmdline.o):(.text+0x6c84): more undefined references to `std::__throw_bad_array_new_length()' follow
       > /nix/store/m2vh2ny7bqpwij1gpmvl5gxj7y4dgr4f-binutils-2.38/bin/ld: /nix/store/33fy01bvz2s4fq7nb536mbvcy1ifdknz-glog-0.6.0/lib/libglog.so: undefined reference to `std::__throw_bad_array_new_length()@GLIBCXX_3.4.29'
       > collect2: error: ld returned 1 exit status
       > make[2]: *** [src/exe/CMakeFiles/colmap_exe.dir/build.make:256: src/exe/colmap] Error 1
       > make[1]: *** [CMakeFiles/Makefile2:803: src/exe/CMakeFiles/colmap_exe.dir/all] Error 2
       > make: *** [Makefile:136: all] Error 2
       For full logs, run 'nix log /nix/store/w6gr8cjy42n5y544nxpfx9vcasm2k6aw-colmap-3.7.drv'.
error: 2 dependencies of derivation '/nix/store/5fiaywf2vncf60gyd4vlpazglmlxiqvb-review-shell.drv' failed to build
```

--- a/pkgs/applications/science/misc/colmap/default.nix
+++ b/pkgs/applications/science/misc/colmap/default.nix
@@ -17,7 +17,7 @@ mkDerivation rec {
      owner = "colmap";
      repo = "colmap";
      rev = version;
-     sha256 = "sha256-uVAw6qwhpgIpHkXgxttKupU9zU+vD0Za0maw2Iv4x+I=";
+     hash = "sha256-uVAw6qwhpgIpHkXgxttKupU9zU+vD0Za0maw2Iv4x+I=";
   };
 
   # TODO: rm once the gcc11 issue is closed, https://github.com/colmap/colmap/issues/1418#issuecomment-1049305256

--- a/pkgs/applications/science/misc/colmap/default.nix
+++ b/pkgs/applications/science/misc/colmap/default.nix
@@ -1,10 +1,14 @@
 { mkDerivation, lib, fetchFromGitHub, cmake, boost17x, ceres-solver, eigen,
   freeimage, glog, libGLU, glew, qtbase,
-  cudaSupport ? false, cudatoolkit ? null }:
+  cudaSupport ? false, cudaPackages }:
 
-assert !cudaSupport || cudatoolkit != null;
+assert cudaSupport -> cudaPackages != { };
 
-let boost_static = boost17x.override { enableStatic = true; };
+let
+  boost_static = boost17x.override { enableStatic = true; };
+
+  # TODO: migrate to redist packages
+  inherit (cudaPackages) cudatoolkit;
 in
 mkDerivation rec {
   version = "3.7";
@@ -15,6 +19,11 @@ mkDerivation rec {
      rev = version;
      sha256 = "sha256-uVAw6qwhpgIpHkXgxttKupU9zU+vD0Za0maw2Iv4x+I=";
   };
+
+  # TODO: rm once the gcc11 issue is closed, https://github.com/colmap/colmap/issues/1418#issuecomment-1049305256
+  cmakeFlags = lib.optionals cudaSupport [
+    "-DCUDA_NVCC_FLAGS=--std=c++14"
+  ];
 
   buildInputs = [
     boost_static ceres-solver eigen

--- a/pkgs/applications/science/misc/colmap/default.nix
+++ b/pkgs/applications/science/misc/colmap/default.nix
@@ -28,7 +28,9 @@ mkDerivation rec {
   buildInputs = [
     boost_static ceres-solver eigen
     freeimage glog libGLU glew qtbase
-  ] ++ lib.optional cudaSupport cudatoolkit;
+  ] ++ lib.optionals cudaSupport [
+    cudatoolkit
+  ];
 
   nativeBuildInputs = [ cmake ];
 

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -12204,8 +12204,8 @@ with pkgs;
 
   colm = callPackage ../development/compilers/colm { };
 
-  colmap = libsForQt5.callPackage ../applications/science/misc/colmap { };
-  colmapWithCuda = colmap.override { cudaSupport = true; cudatoolkit = cudatoolkit_11; };
+  colmap = libsForQt5.callPackage ../applications/science/misc/colmap { cudaSupport = config.cudaSupport or false; };
+  colmapWithCuda = colmap.override { cudaSupport = true; };
 
   chickenPackages_4 = callPackage ../development/compilers/chicken/4 { };
   chickenPackages_5 = callPackage ../development/compilers/chicken/5 { };

